force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,171 @@
+public with sharing class OpportunityTriggerHandler extends TriggerHandler{
+
+        public override void beforeInsert(){
+
+            OpportunityUtilities.defaultNewCustomers(Trigger.new);
+
+            OpportunityUtilities.setNewOppyAsNewCustomer(Trigger.new);
+        }
+        
+        public override void afterInsert(){
+
+            OpportunityUtilities.newOppyGetsTask(Trigger.new);
+
+        }
+    
+        public override void beforeDelete(){
+
+            OpportunityUtilities.preventClosedOppyDelete(Trigger.old);
+
+            // When an opportunity is deleted prevent the deletion of a closed won opportunity if the account industry is 'Banking'.
+            OpportunityUtilities.cantDeleteClosedWonBanking(Trigger.old);
+        }
+
+        public override void beforeUpdate(){
+    
+            OpportunityUtilities.opportunityMustBeFiveThousand(Trigger.new);
+            OpportunityUtilities.appendStageChangeInOppyDescription(Trigger.new);
+            OpportunityUtilities.updatePrimaryContact(Trigger.new);
+
+        }    
+
+        @TestVisible
+        public override void afterUndelete(){
+
+            OpportunityUtilities.assignPrimaryContact((Map<Id, Opportunity>) Trigger.newMap);
+            
+        }
+
+        
+        public override void afterDelete(){
+        
+            OpportunityUtilities.notifyOwnersOpportunityDeleted(Trigger.old);
+        }
+}
+/*
+
+
+
+
+
+
+                // Send email notifications when an Opportunity is deleted 
+                else if (Trigger.isDelete){
+                    notifyOwnersOpportunityDeleted(Trigger.old);
+                } 
+                // Assign the primary contact to undeleted Opportunities
+                else if (Trigger.isUndelete){
+                    assignPrimaryContact(Trigger.newMap);
+                }
```

Comment:
It is good to keep this while you are working on refactoring, but usually before final release to production, I recommend deleting this commented out code.